src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs

```diff
@@ -350,8 +414,28 @@ private void InsertEntry(NodeSearchElementViewModel entry, IEnumerable<string> c
             NodeSearchElementViewModel entry, NodeCategoryViewModel target,
             IEnumerable<string> categoryNames)
         {
-            var newTargets =
-                categoryNames.Select(name => new NodeCategoryViewModel(name));
+            if (!categoryNames.Any())
+            {
+                target.Entries.Add(entry);
+                return;
+            }
+
+            var path = target.FullCategoryName;
+            var newTargets = categoryNames.Select(name =>
+            {
+                path = AddToFullPath(path, name);
+
+                var cat = new NodeCategoryViewModel(name);
+                cat.FullCategoryName = path;
+                return cat;
+            }).ToList();
+
+            // Last category in path is class. Should be created and inserted to collection 
+            // classes container ClassesNodeCategoryViewModel
+            int indexToInsertClass = newTargets.Count - 1;
+            var classParent = indexToInsertClass > 0 ? newTargets[indexToInsertClass - 1] : target;
+            var newClass = new ClassesNodeCategoryViewModel(Configurations.ClassesDefaultName, classParent);
+            newTargets.Insert(indexToInsertClass, newClass);
```

Comment:
For the above chunks of codes, here's what I would recommend:

```csharp
// The last entry 'NodeCategoryViewModel' represents a class. For our example the 
// entries in 'newTargets' are:
// 
//      NodeCategoryViewModel("MyAssembly.MyNamespace")
//      NodeCategoryViewModel("MyAssembly.MyNamespace.MyClass")
// 
// Since all class entries are contained under a 'ClassesNodeCategoryViewModel', 
// we need to create a new 'ClassesNodeCategoryViewModel' instance, and insert it 
// right before the class entry itself to get the following list:
// 
//      NodeCategoryViewModel("MyAssembly.MyNamespace")
//      ClassesNodeCategoryViewModel("Classes")
//      NodeCategoryViewModel("MyAssembly.MyNamespace.MyClass")
// 
int indexToInsertClass = newTargets.Count - 1;
var parentOfClass = indexToInsertClass > 0 ? newTargets[indexToInsertClass - 1] : target;
var newClassCategory = new ClassesNodeCategoryViewModel(...);
newTargets.Insert(indexToInsertClass, newClassCategory);
```